src/vue/useQueries.ts

```diff
@@ -114,33 +123,36 @@ export type UseQueriesResults<
   : // Fallback
     QueryObserverResult[];
 
+type UseQueriesOptionsArg<T extends any[]> = readonly [...UseQueriesOptions<T>];
+
 export function useQueries<T extends any[]>(
-  queries: readonly [...UseQueriesOptions<T>]
+  queries: Ref<UseQueriesOptionsArg<T>> | UseQueriesOptionsArg<T>
 ): Readonly<UseQueriesResults<T>> {
-  const queryClientKey = queries[0]?.queryClientKey;
+  const queryClientKey = (unref(queries) as UseQueriesOptionsArg<T>)[0]
+    ?.queryClientKey;
```

Comment:
I had to do the `as UseQueriesOptionsArg<T>` assertions because typescript apparently can't properly derive the return type of `unref`

It works totally fine in my IDE, no ts failures.

Maybe the failure come from the way Typescript works with Rollup: maybe it can't infer the type from the function and has to rely on the function type (which apparently is not enough)

![image](https://user-images.githubusercontent.com/427621/148817215-0608d520-eb48-4471-a56e-0dac97f2fd1e.png)